### DISCUSSION TITLE
Keymap of file type based is failed when start nvim from terminal at startup

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#2123] [Q&A] [answered]
Here is my setting for markdown file, in the file "/home/lizhe/.config/nvim/ftplugin/markdown.lua". keymap just for markdown file.
It works fine when I just start nvim in terminal with typing `nvim<CR>`. But if start from "alacritty -e nvim", it not work(you can change alacritty to kitty is same). How to fix it?
```lua
vim.cmd([=[
vnoremap <A-m> <Plug>VSurround]gv<Plug>VSurround]
nnoremap <A-m> viW<Plug>(nvim-surround-visual)]viW<Plug>(nvim-surround-visual)]2l
nnoremap <A-y> viW<Esc>a](<C-r>+)<Esc>gvo<Esc>i[<Esc>
vnoremap <A-y> <Esc>a](<C-r>+)<Esc>gvo<Esc>i[<Esc>
]=])
```

**Comment by dpetka2001** [comment]
It must have something to do with the loading order when invoking Neovim with `alacritty/kitty -e`. Unfortunately, this is not a LazyVim issue and you should ask about your problem at the corresponding repo of the terminal you use. What is the reason that you open Neovim with `alacritty -e nvim`?

**Reply by OriLiMu** [reply]
the reason I start nvim in alacrity is in obsidian I write a command to edit the current file in nvim, this command will create a new terminal and start nvim

**Reply by dpetka2001** [reply]
Just open the file directly in Neovim. I don't know what alacritty and kitty does with the `-e` flag under the hood. But this is not a LazyVim or even Neovim issue.

**Reply by OriLiMu** [reply]
"-e" for execute command at start up. 

**Sorry I make some mistake**. 
It seems no relationship with terminal here.

I have two keymap both map key <kbd>A-m</kbd>.  
- One keymap is in the ftplugin folder (markdown keymap)
- Another is in the `/lua/config/keymaps.lua`.(normal keymap) 

If run the command "nvim" in terminal and open a markdown file **in nvim** <kbd>A-m</kbd> works fine(Always in one terminal no new terminal created) as a markdown keymap. 
But if I run the command "nvim test.md"(Directly open a file at terminal). The <kbd>A-m</kbd> will be the (normal map) what I want is the (markdown map here). 

It should be a issue in lazyvim? 

Before I think it related to terminal is I run the command "alacritty -e nvim xxxx.md", I thought it something wrong with terminal. Actually remove the terminal part still have the issue

**Reply by dpetka2001** [reply, marked as answer]
Put a `vim.schedule` around your `vim.cmd`
